### DISCUSSION TITLE
What kind of cryptography is used for encrypted nodes?

### DISCUSSION THREAD

**Discussion by Coderific** [freeplane/freeplane#900] [General]
Assuming a symmetric key, but what algorithm and length?

Has this been assessed by a third party?

Thank you

**Comment by sifran-github** [comment]
i THINK that it is PBEWithMD5AndDES which can be found here:
https://github.com/freeplane/freeplane/blob/1.10.x/freeplane/src/main/java/org/freeplane/features/encrypt/SingleDesEncrypter.java

Code about encryption:
https://github.com/freeplane/freeplane/tree/1.10.x/freeplane/src/main/java/org/freeplane/features/encrypt

**Comment by amariusz** [comment]
In the same folder there's also `TripleDesEncrypter.java` with "PBEWithMD5AndTripleDES". So maybe it's more secure than single DES but still dated.

https://github.com/freeplane/freeplane/blob/1.10.x/freeplane/src/main/java/org/freeplane/features/encrypt/TripleDesEncrypter.java

@dpolivaev maybe it would be worth considering to use some modern drop-in replacement encryption available currently in Java?
I've asked chatgpt and it's suggestion was `PBKDF2WithHmacSHA256`.

Edit.
For backward compatibility old encryption should also be supported.
